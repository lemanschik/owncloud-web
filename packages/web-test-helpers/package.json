{
  "name": "web-test-helpers",
  "private": true,
  "main": "src/index.ts",
  "peerDependencies": {
    "@casl/ability": "^6.3.3",
    "@casl/vue": "^2.2.1",
    "axios": "0.27.2",
    "design-system": "@ownclouders/design-system",
    "vue3-gettext": "^2.3.3",
    "vue-router": "4.1.6",
    "vuex": "4.1.0"
  }
}
