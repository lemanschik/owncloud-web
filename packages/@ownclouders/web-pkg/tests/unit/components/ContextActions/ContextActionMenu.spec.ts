import ContextActionMenu from 'web-pkg/src/components/ContextActions/ContextActionMenu.vue'
import { defaultPlugins, shallowMount } from 'web-test-helpers'

describe('ContextActionMenu component', () => {
  it('renders the menu with actions', () => {
    const menuSections = [{ name: 'action 1' }, { name: 'action 2' }]
    const { wrapper } = getShallowWrapper(menuSections)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.oc-files-context-actions').exists()).toBeTruthy()
    expect(wrapper.findAll('.oc-files-context-actions').length).toEqual(menuSections.length)
  })
})

function getShallowWrapper(menuSections, items = []) {
  return {
    wrapper: shallowMount(ContextActionMenu, {
      props: {
        menuSections,
        actionOptions: { resources: items }
      },
      global: {
        plugins: [...defaultPlugins()]
      }
    })
  }
}
