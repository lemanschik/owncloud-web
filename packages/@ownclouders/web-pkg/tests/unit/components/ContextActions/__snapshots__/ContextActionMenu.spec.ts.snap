// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextActionMenu component renders the menu with actions 1`] = `
<div id="oc-files-context-menu">
  <oc-list-stub class="oc-files-context-actions oc-pb-s oc-files-context-actions-border" id="oc-files-context-actions-action 1" raw="false"></oc-list-stub>
  <oc-list-stub class="oc-files-context-actions oc-pt-s" id="oc-files-context-actions-action 2" raw="false"></oc-list-stub>
</div>
`;
